src/Bridge/Doctrine/Orm/Extension/OrderExtension.php

```diff
@@ -47,10 +47,22 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
             if (null !== $defaultOrder) {
                 foreach ($defaultOrder as $field => $order) {
                     if (is_int($field)) {
+                        // Default direction
                         $field = $order;
                         $order = 'ASC';
                     }
-                    $queryBuilder->addOrderBy('o.'.$field, $order);
+                    if (false === ($pos = strpos($field, '.'))) {
+                        // Configure default filter with property
+                        $field = 'o.'.$field;
+                    } else {
+                        // Configure default filter with association
+                        // Ensure query join is done on association
+                        $associationName = substr($field, 0, $pos);
+                        if (!in_array($associationName, $queryBuilder->getRootAliases(), true)) {
+                            $queryBuilder->join('o.'.$associationName, $associationName);
```

Comment:
Maybe this: https://github.com/api-platform/core/blob/0e1af77ee629b4f3d4841e172a67bfb6eda0db8b/src/Bridge/Doctrine/Orm/Filter/AbstractFilter.php#L382-L401